util/sdk/macros/index.mjs

```diff
@@ -389,11 +417,16 @@ function generateMethods(json = {}, onlyEvents = false) {
         params: getMethodSignatureParams(moduleName, methodObj, { isInterface: false })
       }
 
-      const template = getTemplateForMethod(methodObj)
+      let template = getTemplateForMethod(methodObj)
+      if (isPropertyMethod(methodObj)) {
+        template = getTemplate('methods/polymorphic-property.js')
```

Comment:
Hmmm. I was hoping to keep template -> tag name 1:1.

I was imagining for this, we'd have:

- methods/property/index.js
- methods/property/immutable.js
- methods/property/readyonly.js

And then update getTemplateForMethod to parse out any `:` and use that as the directory name.

Then put simpler code for each type in each of those individual templates.

The way you have it built with the same template for all three adds unused code (and kilobytes) for each property to the final build. I'd rather have the extra stuff in redudant templates to keep the compiled artifact smaller.

Reply:
It doesn't really add extra stuff because it is meant to throw an error to the client (from the SDK without hitting FBOS) if they try and misuse a property.
Also it is using a reusable module I created `Prop`. So each of these methods just calls into the Prop module with the values of the flags. So it isn't using any more space.

Reply:
the template is this:
```
function ${method.name}() {
  return Prop.prop('${info.title}',  '${method.name}', arguments, ${method.property.immutable}, ${method.property.readonly})
}
```
So basically the smallest amount of code generated per property.
I think it would be a lot more if we had a template per tag type.

Reply:
yeah, looking more closely I can see that the way you wrote it is smaller (in kb) than how I suggested.